nni/algorithms/compression/v2/pytorch/pruning/tools/base.py

```diff
@@ -133,7 +133,8 @@ def patched_criterion(output, target):
         super().__init__(compressor)
         self.trainer = trainer
         self.training_epochs = training_epochs
-        self._origin_optimizer = optimizer
+        self._origin_optimizer_cls = optimizer.__class__ if optimizer is not None else None
+        self._origin_optimizer_state_dict = optimizer.state_dict() if optimizer is not None else None
```

Comment:
what cases is this update not able to handle?

Reply:
some fancy usage of optimizer?

Reply:
just don't want to reference the parameters in model...